src/Log/Engine/BaseLog.php

```diff
@@ -140,13 +140,18 @@ protected function _format(string $message, array $context = []): string
             }
 
             if (is_object($value)) {
-                if (method_exists($value, '__toString')) {
-                    $replacements['{' . $key . '}'] = (string)$value;
+                if (method_exists($value, 'toArray')) {
+                    $replacements['{' . $key . '}'] = json_encode($value->toArray(), JSON_UNESCAPED_UNICODE);
                     continue;
                 }
 
-                if (method_exists($value, 'toArray')) {
-                    $replacements['{' . $key . '}'] = json_encode($value->toArray(), JSON_UNESCAPED_UNICODE);
+                if (method_exists($value, '__serialize')) {
```

Comment:
This should be ok since we're implementing both, but __serialize() won't be called for php 7.4.

Reply:
Right, but those classes already implement `Serializable` which is checked earlier. This check for `__serializable` is for future proofing when the check for Serializable will be removed and for app code which only use the magic methods and not implement Serializable.

Reply:
Ah, I see it. The order change confused me.